### DISCUSSION TITLE
Is it possible to do `git clone --depth=1 <repository>` with libgit2?

### DISCUSSION THREAD

**Discussion by nikivdev** [libgit2/libgit2#6526] [Q&A] [answered]
Is it possible to do `git clone --depth=1 https://github.com/libgit2/libgit2` with libgit2?

I found this [open issue](https://github.com/libgit2/libgit2/issues/3058) about sparse checkouts but I am not sure that it's the same.

It also links to [open PR](https://github.com/libgit2/libgit2/pull/6396) that has not yet been merged although @ethomson recently replied that he will look into merging it. ♥️

Someone in the PR comment mentioned that this option is available via a flag but searching on website I didn't find which flag that was.

I hope there is some way to do a `--depth=1` clone without having to build libgit2 from a fork which I want to avoid doing as I don't know C.

Thank you.

**Comment by lrm29** [comment, marked as answer]
The issue you link to (#3508) is for shallow support, not sparse checkouts. Shallow refers to the amount of history, sparse checkout refers to how much of the working folder is checked out.

To use #6396 you'll need to build libgit2 from a fork, until it is merged into libgit2.
